### PR TITLE
BFD-1518: Add v2 mapping for Revenue Status Code (HHA/Outpatient)

### PR DESCRIPTION
**Pull request by lsmitchell**
**JIRA Ticket:**
[BFD-1518](https://jira.cms.gov/browse/BFD-1518)

**User Story or Bug Summary:**

As a peering partner customer, I want Revenue Center Status Indicator Code mapped in V2, so I can use that field.

---

### What Does This PR Do?

- Add mapping for HHA and Outpatient claims so that Revenue Status Code is mapped to the final output
    - REV_CNTR_STUS_IND_CD => ExplanationOfBenefit.item.revenue.extension
- Add testing to cover the new scenario
    - Add test class for transformerUtilsV2 since none existed

I had a thought that this mapping might actually make more sense to do within mapEobCommonItemRevenue, but since this mapping should only affect HHA and Outpatient I wasnt sure if that was a good place to put it, since it would get mapped if the other claim types would happen to have the indicator.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.
* Verify the fields look mapped correctly and consistent with other mapping
* Review if the non-null annotations make sense (It seemed safer to have them)
* Review the new test class and ensure its not too redundant

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [X] Yes
      * [] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* Comms are crafted to communicate the change

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — APPROVED** [review]

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+import java.util.Optional;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.ExplanationOfBenefit;
+import org.hl7.fhir.r4.model.Extension;
+import org.junit.jupiter.api.Test;
+
+/** Tests the utility methods within the {@link TransformerUtilsV2}. */
+public class TransformerUtilsV2Test {
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java:16]: There was no specific test class for transformerUtilsV2, so I made this to at least cover the new code I wrote. While these methods are indirectly covered by the individual mapping tests, we can do more direct testing easier with a specific test class to directly test those methods.

**Comment by lsmitchell** [comment]
Fixing some tests I didn't notice were failing.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java [hunk 2]
```diff
@@ -260,6 +260,9 @@ private static ExplanationOfBenefit transformClaim(HHAClaim claimGroup) {
           C4BBPractitionerIdentifierType.NPI,
           C4BBClaimProfessionalAndNonClinicianCareTeamRole.PERFORMING,
           line.getRevenueCenterRenderingPhysicianNPI());
+
+      // REV_CNTR_STUS_IND_CD => ExplanationOfBenefit.item.revenue.extension
+      TransformerUtilsV2.mapEobCommonItemRevenueStatusCode(item, eob, line.getStatusCode());
```
Comment (bfausett77) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java:265]: Is there a check before this to ensure that line.getStatusCode is not null?

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2.java [hunk 3]
```diff
@@ -402,6 +402,9 @@ private static ExplanationOfBenefit transformClaim(OutpatientClaim claimGroup) {
           C4BBPractitionerIdentifierType.NPI,
           C4BBClaimInstitutionalCareTeamRole.PERFORMING,
           line.getRevenueCenterRenderingPhysicianNPI());
+
+      // REV_CNTR_STUS_IND_CD => ExplanationOfBenefit.item.revenue.extension
+      TransformerUtilsV2.mapEobCommonItemRevenueStatusCode(item, eob, line.getStatusCode());
```
[the remark this answers is not in the document]
Comment (bfausett77) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2.java:407]: Is there a check before this to ensure that line.getStatusCode is not null?

[the remark this answers is not in the document]
Comment (bfausett77) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2.java:407]: Never mind!  I see the check in the transformUtilsV2!

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java (hunk 2 above)
Comment (bfausett77) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java:265]: Never mind!  I see the check in the transformUtilsV2!

**Review by bfausett77 — DISMISSED** [review]
This looks good to me.  I had some questions, but was able to answer them myself as I went through the PR.  Good job!

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 4]
```diff
@@ -3550,6 +3552,33 @@ static ItemComponent mapEobCommonItemRevenue(
...
+   * @param item the item to add the extension to, if the required data is present
+   * @param eob the root eob (only used for logging purposes)
+   * @param statusCode the status code to check for and add data from if exists
+   * @return the {@link ItemComponent}
+   */
+  static ItemComponent mapEobCommonItemRevenueStatusCode(
+      @Nonnull ItemComponent item, @Nonnull ExplanationOfBenefit eob, Optional<String> statusCode) {
+
+    Assert.notNull(item, "Item must be non-null");
+    Assert.notNull(eob, "Eob must be non-null");
```
Comment (christrotteradhoc) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3570]: I like asserting these (an alternative is lombok annotations but it is not widely adopted in BFD outside of PACA -- something that we should discuss when we are working on style guide). My only question is whether there is precedent in other parts of the code for any other null checking assertions that use a different mechanism?

Comment (lsmitchell) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3570]: From a search I don't see any other specific null checking mechanisms outside of general != null checks in the code logic, so this shouldnt conflict with any existing strategies.

**Review by mackec1 — APPROVED** [review]
LGTM.

File: apps/bfd-server/dev/api-changelog.md [hunk 5]
```diff
@@ -1,5 +1,37 @@
 # API Changelog
 
+## BFD-1518
```
Comment (ajsarkar128) [inline apps/bfd-server/dev/api-changelog.md:3]: I'd expand on this title with high level what was changed. Looks like it's written below
